src/neoxp/Node/Modules/ExpressRpcMethods.cs

```diff
@@ -403,6 +405,90 @@ public JObject GetNep11Properties(JArray @params)
             }
             return json;
         }
+        
+        private const byte Prefix_Contract = 8;
+        private const byte Prefix_NextAvailableId = 15;
+
+        private int LastUsedContractId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.TryGet(key);
+            return (int)(BigInteger)item;
+        }
+
+        private void SetLastUsedContractId(DataCache snapshot, int newId)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            item.Set(newId);
+        }
+        
+        private int GetNextAvailableId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            int value = (int)(BigInteger)item;
+            item.Add(1);
+            return value;
+        }
+        
+        [RpcMethod]
+        public JObject? ExpressPersistContract(JObject @params)
+        {
+            var state = @params[0]["state"];
```

Comment:
I considered making it optional to override an existing contract in case the remote side updates the contract and we want to pull the new one. I believe the easier solution is deleting the existing contract (using point 1 of [this](https://github.com/ngdenterprise/neo3-visual-tracker/issues/133) visual tracker idea) and then downloading again. The downside is that it will also drop the storage which could require the user to call any functions that modified the storage again. But at least that should be obvious when the contract is deleted.

Reply:
Would it make sense to make each optional? You know more about this scenario that I do, but it seems like each of these scenarios has value:

* Download remote contract/storage initially
* Update local contract storage from remote chain w/o updating contract
* Update local contract from remote chain w/o updating contract
* Re-download remote contract/storage, deleting existing copy + and contract state (though probably reuse contract id)

Note, whatever direction we go here the options need to be available on the download command itself, not just in the custom RPC method